### PR TITLE
Add support for scraping parental guide page & paginated title akas, episodes & fix plot in reference page

### PR DESCRIPTION
**Pull request by mhdzumair**

### CODE REVIEW

**Comment by mhdzumair** [comment]
@uyar ready for review. Sorry for not able to push the changes sooner.

**Comment by uyar** [comment]
Thanks a lot for your work. I'll get to it as soon as I can.

**Review by uyar — APPROVED** [review]

**Comment by uyar** [comment]
Sorry, I've mistakenly approved all changes. I just meant the commit that fixes the bad plot parser, I've pulled in that one. I'll get to the others too.

**Review by uyar — CHANGES_REQUESTED** [review]
Style checks are failing. You can try with `ruff check --preview cinemagoerng tests`.

File: cinemagoerng/model.py
Comment (uyar) [inline cinemagoerng/model.py:1]: The class name `Votes` can be confused with the movie votes. Maybe rename to something like `AdvisoryVotes`?

File: tests/test_parentalguide.py
Comment (uyar) [inline tests/test_parentalguide.py:1]: Could you add some more tests to cover other cases?

File: cinemagoerng/specs/title_parental_guide.json
Comment (uyar) [inline cinemagoerng/specs/title_parental_guide.json:1]: Indentation is inconsistent. We should stick to 2 spaces.

File: tests/test_parentalguide.py
Comment (mhdzumair) [reply tests/test_parentalguide.py:1]: updated

**Comment by uyar** [comment]
Looks like you've run black on the code but I don't like black. It changed way too many files now that the PR has lost its focus and diverged from its purpose.

File: tests/test_parentalguide.py
Comment (mhdzumair) [reply tests/test_parentalguide.py:1]: Added more test data

**Comment by mhdzumair** [comment]
> Looks like you've run black on the code but I don't like black. It changed way too many files now that the PR has lost its focus and diverged from its purpose.

I usually use black, however I'm not setting up black for this project, instead I ran this command, 
`ruff format cinemagoerng tests`   also updated the max line length to 119, if you want to revert it back the line length i can revert it and reformat it again.

**Comment by uyar** [comment]
> I usually use black, however I'm not setting up black for this project, instead I ran this command, `ruff format cinemagoerng tests` also updated the max line length to 119, if you want to revert it back the line length i can revert it and reformat it again.

Though, AFAIK, rust-format is more configurable than black, I don't like autoformatters. Especially the line length issue. I believe that 79 characters is a good limit (except for tests) and if you have a good reason to break it, you can do so explicitly by using a `noqa: E501`.

The problem with the PR at the moment is that a lot of stuff have been bundled into one large changeset (like the title also indicates). If you have the time, could you divide these into smaller PRs for one thing each (parental guide, paginated akas, etc). I've already separated and merged the plot fix.

If you don't have the time or just don't want to do it, that's fine. I'll go over your changes and incorporate them one by one. But, for future reference, please keep your PRs focused to one thing at a time.

**Comment by mhdzumair** [comment]
Sure will create a seperate PRs.

**Pull request closed without merging**